Assets/Editor/QuickSpriteSettings/SpriteSettingsConfig.cs

```diff
@@ -21,39 +19,84 @@ void OnEnable()
             SettingsSets = SettingsSets ?? new List<SpriteSettings>();
             if (SettingsSets.Count < 1)
             {
-                SettingsSets.Add(new SpriteSettings());
+                AddDefaultSpriteSetting ();
             }
-
-            Settings = SettingsSets[0];
+        }
+        
+        public void AddDefaultSpriteSetting ()
+        {
+            SettingsSets.Add(new SpriteSettings());
         }
     }
 
     [CustomEditor(typeof(SpriteSettingsConfig))]
     public class SpriteSettingsConfigEditor : Editor
     {
-        SerializedProperty settings;
+        private ReorderableList list;
 
-        void Awake()
-        {
-            settings = serializedObject.FindProperty("SettingsSets");
+        private void OnEnable() {
+            list = new ReorderableList(serializedObject, 
+                    serializedObject.FindProperty("SettingsSets"), 
+                    true, true, true, true);
+            list.drawHeaderCallback = (Rect rect) => {  
+                EditorGUI.LabelField(rect, "Saved SpriteSettings");
+            };
         }
 
         public override void OnInspectorGUI()
         {
             serializedObject.Update();
 
-            if (settings.isArray)
-            {
-                settings.Next(true); // generic field
-                settings.Next(true); // array size field
-
-                // first array index
-                settings.Next(true);
-
-                EditorGUILayout.PropertyField(settings);
-            }
-
+            list.DoLayoutList();
+            
+            EditorGUILayout.Space ();
+            EditorGUILayout.LabelField ("Selected Settings", EditorStyles.largeLabel);
+            
+            DrawSelectedSpriteSetting (serializedObject);
+            
             serializedObject.ApplyModifiedProperties();
         }
+        
+        void DrawSelectedSpriteSetting (SerializedObject serializedObject)
+        {
+            int selectedIndex = list.index;
+            
+            SerializedProperty settingsSets = serializedObject.FindProperty ("SettingsSets");
+            
+            if (selectedIndex >= 0) {
+                EditorGUILayout.PropertyField (settingsSets.GetArrayElementAtIndex (selectedIndex));
+            } else {
+                EditorStyles.label.wordWrap = true;
+                if (GetPropertyArraySize (settingsSets) > 0) {
+                    EditorGUILayout.LabelField ("Select a Saved SpriteSetting to edit or remove it.");
+                } else {
+                    EditorGUILayout.LabelField ("No SpriteSettings have been saved. Create a new SpriteSetting from the list above.");
+                }
+            }
+        }
+        
+        int GetPropertyArraySize (SerializedProperty listAsProperty)

```

Comment:
Ugh copied this from another project and it didn't fix the whitespace formatting. I'll try to remember to fix that in a future commit.